packages/i18n/src/create-i18n.js

```diff
@@ -116,6 +145,19 @@ export const createI18n = ( initialData, initialDomain ) => {
 			...DEFAULT_LOCALE_DATA[ '' ],
 			...tannin.data[ domain ][ '' ],
 		};
+
+		listeners.forEach( ( listener ) => listener() );
```

Comment:
Would it make sense to use `doAction` here instead?

Reply:
Something like `doAction( 'i18n.update' )`? That's an interesting idea! :+1:

Reply:
Is this still a to-do?

Reply:
I tried to implement @yuliyan's suggestion with the `doAction` hook, but then found it's not practical to work with. 

With the hook, it's no longer enough to pass an `I18n` instance as a prop the `I18nProvider`. I also need to pass along an instance of the `Hooks` object, the one that was used to construct that `I18n` instance. Because there's no direct way to subscribe to the `I18n` instance, I need to go indirectly through `Hooks`.

Also, what if I have multiple instances of `I18n` created with the same instance of `Hooks`? That's going to happen in the "real-time locale switcher" use case. Then the `doAction` hook would need to receive the `I18n` instance as a parameter, and the callback would need to check if the action fired for our instance, or for someone else.

In the end, I'm finding a `I18n.subscribe` method to be more practical.